opencti-platform/opencti-front/src/private/components/settings/users/UserCreation.jsx

```diff
@@ -51,20 +52,40 @@ const userValidation = (t) => Yup.object().shape({
   confirmation: Yup.string()
     .oneOf([Yup.ref('password'), null], t('The values do not match'))
     .required(t('This field is required')),
+  user_confidence_level_enabled: Yup.boolean(),
+  user_confidence_level: Yup.number()
+    .min(0, t('The value must be greater than or equal to 0'))
+    .max(100, t('The value must be less than or equal to 100'))
+    .when('user_confidence_level_enabled', {
+      is: true,
+      then: (schema) => schema.required(t('This field is required')).nullable(),
+      otherwise: (schema) => schema.nullable(),
+    }),
```

Comment:
a technical field (not to submit) associated with a `when`. Very useful for validation conditioned on other values!